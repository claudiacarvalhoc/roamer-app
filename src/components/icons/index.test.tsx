import React from 'react';
import Icons, { IconsProps } from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const getShallowWrapper = (props) => {
    return shallow(<Icons {...props} />);
};

describe('<Icons />', () => {
    it('render correctly', () => {
        const props: IconsProps = {
            className: 'test',
            types: ['up', 'down', 'check', 'user', 'rate', 'camera', 'book'],
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly when types are not defined', () => {
        const props: IconsProps = {
            className: 'test',
            types: null,
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
});
