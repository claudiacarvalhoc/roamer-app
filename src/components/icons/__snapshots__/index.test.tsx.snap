// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icons /> render correctly 1`] = `
<div
  className="test"
>
  <UpIcon
    className="icon"
    key="type-up"
  />
  <DownIcon
    className="icon"
    key="type-down"
  />
  <CheckIcon
    className="icon"
    key="type-check"
  />
  <ProfileIcon
    className="icon"
    key="type-user"
  />
  <RateIcon
    className="icon"
    key="type-rate"
  />
  <CameraIcon
    className="icon"
    key="type-camera"
  />
  <BookIcon
    className="icon"
    key="type-book"
  />
</div>
`;

exports[`<Icons /> render correctly when types are not defined 1`] = `
<div
  className="test"
/>
`;
